ESSArch_Core/test_util.py

```diff
@@ -173,3 +177,155 @@ def test_get_schema_from_file_with_bad_input(self):
+        self.datadir = os.path.join(self.root, "datadir")
+        self.textdir = os.path.join(self.datadir, "textdir")
+
+        try:
+            os.makedirs(self.textdir)
+        except OSError as e:
+            if e.errno != 17:
+                raise
+
+    def tearDown(self):
+        shutil.rmtree(self.datadir)
+
+    def create_files(self):
+        files = []
+        for i in range(3):
+            fname = os.path.join(self.textdir, '%s.txt' % i)
+            with open(fname, 'w') as f:
+                f.write('%s' % i)
+            files.append(fname)
+
+        return files
+
+    def create_archive_file(self, archive_format):
+        self.create_files()
+
+        output_filename = "archive_file"
+        archive_file_full_path = os.path.join(self.datadir, output_filename)
+
+        return shutil.make_archive(archive_file_full_path, archive_format, self.textdir)
+
+    def create_paginator(self):
+        return None
+
+    def test_list_files_dir_with_default_args_should_raise_NotFound(self):
+        path = self.datadir
+        self.create_files()
+
+        with self.assertRaises(NotFound):
+            list_files(path)
+
+    def test_list_files_tarfile_with_default_args_should_return_response(self):
+        file_path = self.create_archive_file('tar')
+        resp = list_files(file_path)
+        self.assertEqual(resp.status_code, 200)
+        file_names = ["./0.txt", "./1.txt", "./2.txt"]  # TODO: bug in shutil for tar is adding an extra './'
+
+        for el in resp.data:
+            data_name = el['name']
+            data_type = el['type']
+            data_size = el['size']
+            data_modified = el['modified']
+
+            self.assertIn(data_name, file_names)
+            file_names.remove(data_name)
+            self.assertEqual(data_type, "file")
+            self.assertEqual(data_size, 1)
+            self.assertEqual(type(data_modified), datetime.datetime)
+
+    def test_list_files_zip_file_with_default_args_should_return_response(self):
+        file_path = self.create_archive_file('zip')
+        resp = list_files(file_path)
+        self.assertEqual(resp.status_code, 200)
+        file_names = ["0.txt", "1.txt", "2.txt"]
+
+        for el in resp.data:
+            data_name = el['name']
+            data_type = el['type']
+            data_size = el['size']
+            data_modified = el['modified']
+
+            self.assertIn(data_name, file_names)
+            file_names.remove(data_name)
+            self.assertEqual(data_type, "file")
+            self.assertEqual(data_size, 1)
+            self.assertEqual(type(data_modified), datetime.datetime)
+
+    def test_list_files_path_to_file_in_tar(self):
+        file_path = self.create_archive_file('tar')
+        new_folder = os.path.join(file_path, "./0.txt")  # TODO: bug in shutil for tar is adding an extra './'
+
+        resp = list_files(new_folder)
+        self.assertEqual(resp.status_code, 200)
```

Comment:
This is now testing the result of `generate_file_response`. Instead, mock `generate_file_response` and ensure that it is called with the correct arguments. See the following tests for an example: https://github.com/ESSolutions/ESSArch_Core/blob/b01689ebcc2faddbd32664039563f0e77622cb62/ESSArch_Core/WorkflowEngine/tests/test_tasks.py#L37